### ISSUE TITLE
No such S3 Bucket

### ISSUE THREAD

**Issue by DarshanVirdi** [aws/amazon-chime-sdk-js#2042]
Hi,

I have used the following command to setup the demo on my AWS account:

_>> node ./deploy.js -r ap-southeast-1 -b chime-bucket-0210221500 -s chime-stack-0210221500 -a meeting_

During deployment I have noticed that in the script somehow the bucket name goes missing.

        Deploying with following values
        ===============================
        Stack name                   : chime-stack-0210221500
        Region                       : ap-southeast-1
        Confirm changeset            : False
        Deployment s3 bucket         : None
        Capabilities                 : ["CAPABILITY_IAM"]
        Parameter overrides          : {"Region": "ap-southeast-1", "UseChimeSDKMeetings": "true", "UseEventBridge": "false", "ChimeEndpoint": "https://service.chime.aws.amazon.com", "ChimeServicePrincipal": "chime.amazonaws.com"}
        Signing Profiles             : {}

After the deployment is completed, all features of the demo are working, other than the "Cloud Recording"

When I check the Cloud Watch Console I see the following error:

_2022-02-21T01:30:28.007Z	6b68336e-29a1-4e0a-b3c9-d962baa2e36f	ERROR	Invoke Error 	
{
    "errorType": "BadRequest",
    "errorMessage": "No such S3 Bucket: -ap-southeast-1",
    "code": "BadRequest",
    "message": "No such S3 Bucket: -ap-southeast-1",
    "time": "2022-02-21T01:30:27.997Z",
    "requestId": "8d0cf310-3941-4531-9edf-7fa75dc63ea4",
    "statusCode": 400,
    "retryable": false,
    "retryDelay": 20.310368249013443,
    "stack": [
        "BadRequest: No such S3 Bucket: -ap-southeast-1",
        "    at Object.extractError (/var/task/node_modules/aws-sdk/lib/protocol/json.js:52:27)",
        "    at Request.extractError (/var/task/node_modules/aws-sdk/lib/protocol/rest_json.js:49:8)",
        "    at Request.callListeners (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:106:20)",
        "    at Request.emit (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:78:10)",
        "    at Request.emit (/var/task/node_modules/aws-sdk/lib/request.js:686:14)",
        "    at Request.transition (/var/task/node_modules/aws-sdk/lib/request.js:22:10)",
        "    at AcceptorStateMachine.runTo (/var/task/node_modules/aws-sdk/lib/state_machine.js:14:12)",
        "    at /var/task/node_modules/aws-sdk/lib/state_machine.js:26:10",
        "    at Request.<anonymous> (/var/task/node_modules/aws-sdk/lib/request.js:38:9)",
        "    at Request.<anonymous> (/var/task/node_modules/aws-sdk/lib/request.js:688:12)"
    ]
}_

Even though the bucket is created as expected and available, but somehow the bucket name is lost during the deployment.

Can you please guide what changes are required for this to work?

Thank you,
Darshan

**Comment by xuesichao** [comment]
Hi @DarshanVirdi . 

TLDR:
The command you used to deploy the demo is lack of name for S3 bucket used for media capture service.
You correct command should be:
```
node ./deploy.js -r ap-southeast-1 -b chime-bucket-0210221500 -o <name-for-media-capture-s3-bucket> -s chime-stack-0210221500 -a meeting
```
---
If you check the `deploy.js` script: 
https://github.com/aws/amazon-chime-sdk-js/blob/db36d8a4191ad6fe58af68d51008c12a5b1286a9/demos/serverless/deploy.js#L37-L52
`-o` means the prefix for S3 bucket name for media capture service.

And here:
https://github.com/aws/amazon-chime-sdk-js/blob/db36d8a4191ad6fe58af68d51008c12a5b1286a9/demos/serverless/deploy.js#L255-L258
If there's `captureOutputPrefix` then it creates the S3 buckets

And then:
https://github.com/aws/amazon-chime-sdk-js/blob/db36d8a4191ad6fe58af68d51008c12a5b1286a9/demos/serverless/deploy.js#L115-L117
Only when there's `-o` in the deploy command, the `captureOutputPrefix` is not empty.

**Comment by xuesichao** [comment]
You could also refer to this doc:
https://github.com/aws/amazon-chime-sdk-js/tree/main/demos/serverless#media-capture

**Comment by xuesichao** [comment]
The above explanation should help solve your problem. Let me close this issue for now.
If you are still having this problem, please feel free to ping me and I will reopen the issue.

**Comment by DarshanVirdi** [comment]
Hi Xuesichao,

Thank you for your reply, now I understand the mistake I made.
Also thank you for sharing the list of parameters, will help explore the sample further more.

Appreciate your support.

Thank you,
Darshan
